File: Amber.Music/Amber.Music.Domain/ArtistWorkReport.cs
using System;

namespace Amber.Music.Domain
{
    public class ArtistWorkReport
    {
        public Guid ArtistId { get; set; }

        public string Name { get; set; }

        public int TotalSongs { get; set; } = 0;

        public int SongsConsidered { get; set; } = 0;

        public int AverageWords { get; set; } = 0;

        public double Variance { get; set; } = 0;

        public double StandardDeviation { get; set; } = 0;

        public ArtistWork MinWords { get; set; }

        public ArtistWork MaxWords { get; set; }

        public DateTime LastGenerated { get; } = DateTime.Now;

        public DateTime LastAccessed { get; set; } = DateTime.Now;
    }
}

File: Amber.Music/amber-music/src/domain/artist-work-report.ts
import { ArtistWork } from "./artist-work";

export class ArtistWorkReport {
    artistId: string;

    name: string;

    totalSongs: number;

    songsConsidered: number;

    averageWords: number;

    variance: number;

    standardDeviation: number;

    minWords: ArtistWork;

    maxWords: ArtistWork;
}